File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    num_tiles = brown + yellow
    for i in range(1, yellow+1):
        if yellow % i == 0:
            h = i
            w = yellow // i
            if 2 * (w + h + 2) == brown:
                print(f"w : {w} , h : {h}")
                return [w+2, h+2]

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
 public int[] solution(int brown, int yellow) {
        int[] answer = {0, 0};
        for (int i = 1; i <= yellow; i++) {
            if (yellow % i == 0) {
                int w = i;
                int h = yellow / i;
                if ((2 * w + 2 * h + 4) == brown) {
                    answer[0] = w + 2;
                    answer[1] = h + 2;
                }
            }
        }
        return answer;
    }
}